compiler/one-cmds/onelib/argumentparse.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from types import SimpleNamespace
+from typing import List, Tuple
+import shutil
+
+import onelib.backends as backends
+import onelib.utils as oneutils
+
+
+class Action():
+    pass
+
+
+class DriverName(Action):
+    pass
+
+
+class NormalOption(Action):
+    pass
+
+
+class TargetOption(Action):
+    pass
+
+
+class ArgumentParser():
```

Comment:
For the reviewers, this class is used like below.
```python
from onelib import argumentparse
from onelib.argumentparse import DriverName, NormalOption, TargetOption


def command_schema():
    parser = argumentparse.ArgumentParser()
    parser.add_argument("dummy-compile", action=DriverName)
    parser.add_argument("--target", action=TargetOption)
    parser.add_argument("--DSP-quota", action=NormalOption)
    parser.add_argument("-o", action=NormalOption)
    parser.add_argument("input", action=NormalOption)

    return parser

```

Reply:
As a lighter note, I tried to name the library `argparse`, but it was error-prone because the name is same with the standard library - argparse. :cry: